pkg/crc/machine/generate_bundle.go

```diff
@@ -0,0 +1,152 @@
+	if err != nil {
+		return errors.Wrap(err, "Error creating the ssh client")
+	}
+	defer sshRunner.Close()
+	if err := cluster.RemovePullSecretFromTheInstanceDisk(sshRunner); err != nil {
+		return err
+	}
+	ocConfig := oc.UseOCWithSSH(sshRunner)
+	if err := cluster.RemovePullSecretFromCluster(ocConfig); err != nil {
+		return err
+	}
+
+	// Stop the cluster
+	currentState, err := client.Stop()
+	if err != nil {
+		return err
+	}
+	if currentState != state.Stopped {
+		return fmt.Errorf("VM is not stopped, current state is %s", currentState.String())
+	}
+
+	// Create a tmp directory with name of <bundle>_custom
+	tmpDir, err := ioutil.TempDir("", "crc_custom_bundle")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmpDir)
+
+	// Create the custom bundle directory which is used as top level directory for tarball during compression
+	currentBundleName := strings.TrimSuffix(crcBundleMetadata.GetBundleName(), filepath.Ext(crcBundleMetadata.GetBundleName()))
+	customBundleName := fmt.Sprintf("%s_custom", currentBundleName)
+	customBundleDir := filepath.Join(tmpDir, customBundleName)
+	if err := os.Mkdir(customBundleDir, 0775); err != nil {
+		return err
+	}
+
+	// Copy the kubeconfig and kubeadmin-password from bundle to tmpDir
+	kubeConfigFilePath := filepath.Join(customBundleDir, "kubeconfig")
+	err = crcos.CopyFileContents(crcBundleMetadata.GetKubeConfigPath(),
+		kubeConfigFilePath,
+		0640)
+	if err != nil {
+		return fmt.Errorf("error copying kubeconfig file to %s: %v", kubeConfigFilePath, err)
+	}
+
+	kubeConfigAdminPasswordPath := filepath.Join(customBundleDir, "kubeadmin-password")
+	kubePassword, err := crcBundleMetadata.GetKubeadminPassword()
+	if err != nil {
+		return err
+	}
+	err = ioutil.WriteFile(kubeConfigAdminPasswordPath, []byte(kubePassword), 0600)
+	if err != nil {
+		return fmt.Errorf("error writing to %s: %v", kubeConfigFilePath, err)
+	}
+
+	// Copy ssh keyfile
+	sshKeyPath := filepath.Join(customBundleDir, "id_ecdsa_crc")
+	generatedKey := filepath.Join(constants.MachineInstanceDir, constants.DefaultName, "id_ecdsa")
+	if err := crcos.CopyFileContents(generatedKey, sshKeyPath, 0400); err != nil {
+		return err
+	}
+
+	// Copy the oc binary
+	ocFilePath := filepath.Join(constants.CrcOcBinDir, constants.OcExecutableName)
+	if err := crcos.CopyFileContents(ocFilePath, filepath.Join(customBundleDir, constants.OcExecutableName), 0755); err != nil {
+		return err
+	}
+
+	// Copy disk image
+	logging.Infof("Copy the disk image to %s", customBundleDir)
+	if err := copyDiskImage(customBundleDir, crcBundleMetadata); err != nil {
+		return err
+	}
+
+	// Get size of newly created disk
+	diskPath := filepath.Join(customBundleDir, filepath.Base(crcBundleMetadata.GetDiskImagePath()))
+	stat, err := os.Stat(diskPath)
+	if err != nil {
+		return err
+	}
+
+	// update bundle info
+	crcBundleMetadata.Type = "custom"
```

Comment:
not sure I would change `Type`, might be best to add a new `buildInfo.Custom = "true"` field or such?

Reply:
Sure, anyway we are not consuming that info so I will just remove it and then if required we can add it in follow up PR's.

Reply:
I'd really like to have this so that we can detect modified bundles. A file is too easy to rename, especially if this means crc picks it up automatically when it has the correct name.